v2/client/client.go

```diff
@@ -123,6 +121,11 @@ func (c *ceClient) Send(ctx context.Context, e event.Event) protocol.Result {
 	if err = e.Validate(); err != nil {
 		return err
 	}
+
+	// Event has been defaulted and validated, record we are going to preform send.
+	ctx, cb := c.observabilityService.RecordSendingEvent(ctx, e)
```

Comment:
If you move it down here, you don't record failed validation. Is that what you want?

Reply:
I moved this because if we do not do the defaulting first, the recorder does not see the true event on the wire and you do not get to record the `id`

Reply:
Makes sense, maybe in future we could have another signal for the validation failure to provide to the obs service, but that's fine now